app/controllers/concerns/mission_control/jobs/job_filters.rb

```diff
@@ -9,10 +9,18 @@ module MissionControl::Jobs::JobFilters
 
   private
     def set_filters
-      @job_filters = { job_class_name: params.dig(:filter, :job_class_name).presence, queue_name: params.dig(:filter, :queue_name).presence }.compact
+      @job_filters = { job_class_name: params.dig(:filter, :job_class_name).presence, queue_name: params.dig(:filter, :queue_name).presence,
+                       finished_at: date_with_time_zone(params.dig(:filter, :finished_at_start))..date_with_time_zone(params.dig(:filter, :finished_at_end)) }.compact
```

Comment:
there is something left to consider, this method will set finished_at as a begin less and endless range, which will get passed as argument to the `JobsRelation#where` method (https://github.com/rails/mission_control-jobs/blob/a3f959416463ac16428f64784cfd24f56e418de8/app/controllers/mission_control/jobs/jobs_controller.rb#L31) every time so we should fix it

any ideas on how to do it?

Reply:
Maybe something like this? 
```ruby
finished_at: finished_at_range_params) }.compact
# ...

def finished_at_range_params
  range_start, range_end = params.dig(:filter, :finished_at_start), params.dig(:filter, :finished_at_end)
  if range_start || range_end
    (parse_with_time_zone(range_start)..parse_with_time_zone(range_end))
  end
end

def parse_with_time_zone(datetime)
  Time.zone.parse(datetime) if datetime.present?
end
```

Then below you'd need to change this to not break if `@job_filters[:finished_at]` is `nil`, like
```ruby
@job_filters[:finished_at]&.begin
@job_filters[:finished_at]&.end
```

Reply:
awesome, thank u :)